tests/e2e/flowtest/test_maintenance_pod.py

```diff
@@ -0,0 +1,173 @@
+import logging
+import time
+from ocs_ci.ocs import constants
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_ocs_version,
+    bugzilla,
+    tier2,
+    e2e,
+)
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.ocs.resources.deployment import (
+    get_osd_deployments,
+    get_deployments_having_label,
+    get_mon_deployments,
+)
+from ocs_ci.ocs.resources.pod import (
+    get_pods_having_label,
+    Pod,
+    wait_for_pods_to_be_running,
+)
+from ocs_ci.utility.utils import ceph_health_check
+
+logger = logging.getLogger(__name__)
+
+
+@tier2
+@e2e
+@bugzilla("2103256")
+@skipif_ocs_version("<4.12")
+class TestMaintenancePod(E2ETest):
```

Comment:
You are not handling teardown for this test, if the test fails in the middle there is a chance that the deployment pods will be in scale downed state and the subsequent tests in the suite will get skipped as the cluster is not healthy.

Reply:
`def test_maintenance_pod_for_osd(self, ceph_objectstore_factory):`
Here the teardown is handled in the fixture factory

Reply:
same for both the tests. hence in case of failure it is made sure that the debug mode is removed and deployments are scaled up!